WordPress/gutenberg
# Typography: Switch to ToolsPanel for block support UI

Depends on:
- ~https://github.com/WordPress/gutenberg/pull/32392~
- ~https://github.com/WordPress/gutenberg/pull/34725~
- https://github.com/WordPress/gutenberg/pull/35415

Related: 
- https://github.com/WordPress/gutenberg/issues/27331
- https://github.com/WordPress/gutenberg/pull/33743
- https://github.com/WordPress/gutenberg/pull/34027

## Description

This PR:
- Adds new `typography` InspectorControls group slot
- Updates the typography block support hooks to provide callbacks required for ToolsPanel
- Updates the typography hook to render the typography support controls via the new InspectorControls slot

### Default Controls

_This PR only addresses the switch to the new `ToolsPanel` component. Deciding which controls should display by default for which blocks will be addressed via a separate PR._

## How has this been tested?

Manually.

#### Test instructions

1. Edit a post, add a navigation block, then select that block.
2. Confirm that the typography controls within the sidebar are within a ToolsPanel
3. Confirm each typography control works appropriately (Note: this PR does not make any specific block's typography controls "default" in terms of always displaying within the new panel)

## Screenshots 

https://user-images.githubusercontent.com/60436221/138071083-96932e63-5796-471b-90fd-44abec38da6a.mp4

## Types of changes
New feature.

## Checklist:
